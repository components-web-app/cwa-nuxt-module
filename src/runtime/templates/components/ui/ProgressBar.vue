<template>
  <Transition
    enter-from-class="cwa:transform cwa:opacity-0 cwa:w-0"
    enter-active-class="cwa:duration-200 cwa:ease-out"
    enter-to-class="cwa:opacity-100"
    leave-from-class="cwa:opacity-100"
    leave-active-class="cwa:duration-300 cwa:ease-in cwa:delay-200"
    leave-to-class="cwa:transform cwa:opacity-0"
  >
    <div
      v-show="props.show"
      class="cwa:w-full cwa:h-0.5"
    >
      <div
        :class="[colourClass]"
        class="cwa:h-0.5 cwa:transition-all cwa:duration-300"
        :style=" { width: `${props.percent}%` }"
      />
    </div>
  </Transition>
</template>

<script setup lang="ts">
const props = withDefaults(defineProps<{
  show: boolean
  percent: number
  colourClass?: string
}>(), { colourClass: 'cwa:bg-green' })
</script>
