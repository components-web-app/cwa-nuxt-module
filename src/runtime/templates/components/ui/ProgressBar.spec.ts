// @vitest-environment happy-dom
import { describe, expect, test } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import ProgressBar from './ProgressBar.vue'

function createWrapper(percent?: number) {
  return shallowMount(ProgressBar, {
    props: {
      show: percent !== undefined,
      percent: percent || 0,
    },
  })
}
describe('ProgressBar', () => {
  describe('snapshots', () => {
    test('should match snapshot IF progress bar is shown', () => {
      const wrapper = createWrapper(0)

      expect(wrapper.element).toMatchSnapshot()
    })

    test('should match snapshot IF progress bar is NOT shown', () => {
      const wrapper = createWrapper()

      expect(wrapper.element).toMatchSnapshot()
    })

    test('should match snapshot IF progress bar has percentage', () => {
      const wrapper = createWrapper(75)

      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
