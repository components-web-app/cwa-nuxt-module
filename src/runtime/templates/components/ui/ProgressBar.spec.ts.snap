// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProgressBar > snapshots > should match snapshot IF progress bar has percentage 1`] = `
<transition-stub
  appear="false"
  css="true"
  enteractiveclass="cwa:duration-200 cwa:ease-out"
  enterfromclass="cwa:transform cwa:opacity-0 cwa:w-0"
  entertoclass="cwa:opacity-100"
  leaveactiveclass="cwa:duration-300 cwa:ease-in cwa:delay-200"
  leavefromclass="cwa:opacity-100"
  leavetoclass="cwa:transform cwa:opacity-0"
  persisted="true"
>
  <div
    class="cwa:w-full cwa:h-0.5"
  >
    <div
      class="cwa:bg-green cwa:h-0.5 cwa:transition-all cwa:duration-300"
      style="width: 75%;"
    />
  </div>
</transition-stub>
`;

exports[`ProgressBar > snapshots > should match snapshot IF progress bar is NOT shown 1`] = `
<transition-stub
  appear="false"
  css="true"
  enteractiveclass="cwa:duration-200 cwa:ease-out"
  enterfromclass="cwa:transform cwa:opacity-0 cwa:w-0"
  entertoclass="cwa:opacity-100"
  leaveactiveclass="cwa:duration-300 cwa:ease-in cwa:delay-200"
  leavefromclass="cwa:opacity-100"
  leavetoclass="cwa:transform cwa:opacity-0"
  persisted="true"
>
  <div
    class="cwa:w-full cwa:h-0.5"
    style="display: none;"
  >
    <div
      class="cwa:bg-green cwa:h-0.5 cwa:transition-all cwa:duration-300"
      style="width: 0%;"
    />
  </div>
</transition-stub>
`;

exports[`ProgressBar > snapshots > should match snapshot IF progress bar is shown 1`] = `
<transition-stub
  appear="false"
  css="true"
  enteractiveclass="cwa:duration-200 cwa:ease-out"
  enterfromclass="cwa:transform cwa:opacity-0 cwa:w-0"
  entertoclass="cwa:opacity-100"
  leaveactiveclass="cwa:duration-300 cwa:ease-in cwa:delay-200"
  leavefromclass="cwa:opacity-100"
  leavetoclass="cwa:transform cwa:opacity-0"
  persisted="true"
>
  <div
    class="cwa:w-full cwa:h-0.5"
  >
    <div
      class="cwa:bg-green cwa:h-0.5 cwa:transition-all cwa:duration-300"
      style="width: 0%;"
    />
  </div>
</transition-stub>
`;
